framework/bee-py/bee_agent/tools/mcp_tools.py

```diff
@@ -0,0 +1,181 @@
+    tool: MCPToolInfo
+
+
+class MCPToolOutput:
+    """Output class for MCP Tool results."""
+
+    def __init__(self, result: CallToolResult):
+        self.result = result
+
+
+class MCPTool(Tool[MCPToolOutput]):
+    """Tool implementation for Model Context Protocol."""
+
+    def __init__(self, client: ClientSession, tool: MCPToolInfo, **options):
+        """Initialize MCPTool with client and tool configuration."""
+        super().__init__(options)
+        self.client = client
+        self._tool = tool
+        self._name = tool.name
+        self._description = (
+            tool.description
+            or "No available description, use the tool based on its name and schema."
+        )
+        self.emitter = BeeEventEmitter()
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @property
+    def description(self) -> str:
+        return self._description
+
+    def input_schema(self) -> str:
+        return self._tool.inputSchema
+
+    async def _run(
+        self, input_data: Any, options: Optional[Dict] = None
+    ) -> MCPToolOutput:
+        """Execute the tool with given input."""
+        print(f"Executing tool {self.name} with input: {input_data}")  # Debug
+        result = await self.client.call_tool(name=self.name, arguments=input_data)
+        print(f"Tool result: {result}")  # Debug
+        return MCPToolOutput(result)
+
+    @classmethod
+    async def from_client(cls, client: ClientSession) -> List["MCPTool"]:
+        tools_result = await client.list_tools()
+        return [cls(client=client, tool=tool) for tool in tools_result.tools]
+
+
+if __name__ == "__main__":
+    import asyncio
+    from mcp.server import Server
+    from mcp.shared.memory import create_connected_server_and_client_session
+
+    ab_input_schema = {
+        "type": "object",
+        "properties": {"a": {"type": "number"}, "b": {"type": "number"}},
+        "required": ["a", "b"],
+    }
+
+    async def add_tool(
+        name: str, arguments: dict
+    ) -> Sequence[TextContent | ImageContent | EmbeddedResource]:
+        print(f"Add tool called with arguments: {arguments}")  # Debug
+        a, b = float(arguments["a"]), float(arguments["b"])
+        result = str(
+            int(a + b)
+        )  # Convert to int since we're testing with whole numbers
+        print(f"Add tool result: {result}")  # Debug
+        return [TextContent(type="text", text=result)]
+
+    async def multiply_tool(
+        name: str, arguments: dict
+    ) -> Sequence[TextContent | ImageContent | EmbeddedResource]:
+        print(f"Multiply tool called with arguments: {arguments}")  # Debug
+        a, b = float(arguments["a"]), float(arguments["b"])
+        result = str(
+            int(a * b)
+        )  # Convert to int since we're testing with whole numbers
+        print(f"Multiply tool result: {result}")  # Debug
+        return [TextContent(type="text", text=result)]
+
+    async def run_tests():
```

Comment:
should we split this out into the test directory `framework/bee-py/tests` ?

Reply:
yes. We can do it, I'm working is an agent to generate/extract the test and add them in the tests/ folder. 

@vabarbosa we can started to move all the tests there. Works for you?

Reply:
yes, we should

Reply:
ok, let me start with this one.

Reply:
done:

`pytest tests/bee_agent/tools/test_mcp_tool.py `